web/static/js/snips.js

```diff
@@ -86,27 +91,28 @@ const initHeaderObserver = () => {
 
   const observer = new IntersectionObserver(
     ([entry]) => {
-      if (!entry.isIntersecting) {
-        element.removeAttribute("data-hide");
-      } else {
-        element.setAttribute("data-hide", "");
-      }
+      element.toggleAttribute("data-hide", entry.isIntersecting);
     },
     // https://stackoverflow.com/a/61115077
     { rootMargin: "-1px 0px 0px 0px", threshold: [1] }
   );
 
   observer.observe(nav);
+
+  // do not remove the hightlighted lines when scroll to the top
+  element.addEventListener("click", (event) => {
+    event.preventDefault();
+    window.scrollTo({ top: 0, behavior: "smooth" });
+  });
 };
 
 window.addEventListener("hashchange", highlightLines);
-window.addEventListener("DOMContentLoaded", async () => {
+window.addEventListener("DOMContentLoaded", () => {
   initHeaderObserver();
   watchForShiftClick();
   highlightLines();
+  scrollToLine();
 
   mermaid.initialize({ startOnLoad: false, theme: "dark" });
-  await mermaid.run({
-    querySelector: "code.language-mermaid",
-  });
+  mermaid.run({ querySelector: "code.language-mermaid" });
```

Comment:
Is there a specific reason why you removed the `await`? It's an async function, and in the mermaid docs examples it's awaited: https://mermaid.js.org/config/usage.html#using-mermaid-run

Reply:
eh, it's fine as-is. i'll merge and sneak this into the latest release